blog/modules/ROOT/pages/sort-antora-components.adoc

```diff
@@ -0,0 +1,196 @@
+
+.componentSorter.js (Simple configurable sorting)
+[source,javascript]
+----
+include::example$sort-components/componentSorter-simple.js[]
+----
+
+And, finally for this section, an example with a wildcard and explicit "hide" instructions.
+(This is heavily influenced by Dan's link:{ms-sorter-url}[example]).
+
+.componentSorter.js (with component sorting and hiding, and wildcards)
+[source,javascript]
+----
+include::example$sort-components/componentSorter.js[]
+----
+
+These implementations are going to be very noisy if you specify a component that isn't present or leave out a component (in the last implementation).
+Leaving out the console logging or only logging once is an easy solution.
+The component-version sorter has a more sophisticated solution.
+
+== Sorting Versions
+
+Versions occur in two places in the UI: the component explorer and the page version selector.
+Thus we need to modify two partials handlebars templates.
+It would be most convenient to have two helpers, one for each sorting task, but since helpers cannot require each other due to how they are loaded from a virtual file system, we need to get both sorters into the same helper.
+We'll do this by specifying the type of sort as the first parameter.
+
+=== Specifying version sort order
+
+Since there's only one component descriptor for each component-version, we can do version sorting by putting a key in each component descriptor.
+This strategy doesn't work well for components, since the key would have to be the same for every version of the component.
+Another possible version sorting strategy, that we will not pursue, would be to have a global complex sorting key, where each component name is associated with an explicit version sort order.
+For instance....
+
+[source,yaml]
+----
+asciidoc:
+  attributes:
+    page-component-sort-order: component-a[v3,v1,vx],component-c[v1,vy,va],component-b[z,y,x]
+----
+
+Implementing this is left as an exercise for the reader.
+
+To get back to what we will demonstrate, we'll have an asciidoc attribute in each component descriptor with a string-valued sort key:
+
+[source,yaml]
+----
+name: component-a
+version: 1.0
+title: My Fabulous A
+display-version: Version One Point Zero
+
+asciidoc:
+  attributes:
+    version-sort-key: zanzibar
+----
+
+Since we'll be getting the `version-sort-key` value from the version object rather than the page attributes exposed in the UI model, the key name doesn't need to start with the `page-` prefix.
+
+=== Handlebars template modifications
+
+We're already supplying all the necessary information to the sorting helper, so the `nav-explore.hbs` handlebars template doesn't need significant modification.
+In order to avoid modifying the existing component objects, we'll copy them, so the component comparison code needs to be a little more careful, comparing name rather than objects.
```

Comment:
```suggestion
In order to avoid modifying the existing component objects, we'll copy them, so the component comparison code needs to be a little more careful, comparing names rather than objects.
```

Not sure about it but since objects is plural, I'm guessing that names should be plural as well?